### DISCUSSION TITLE
Switching agent types

### DISCUSSION THREAD

**Discussion by patrickshortall26** [JoelForamitti/agentpy#69] [Q&A]
Hi Joël,

First of all, thank you very much! Your package got me through my masters project on a problem in swarm robotics. 

My question is about switching agent types.

In my model I have three types of agents: Healthy, Faulty and Granuloma. 

Each agent exhibits different behaviour based on their type and so I opted to create different classes for each of them rather than have an attribute which defines the type as in your virus spread model. My problem is that I need agents to be able to switch their type based on certain conditions. The agents don't change their attributes when they switch, just their behaviour. For example, their position stays the same, but the algorithm which they use for updating their velocity changes. 

How is the best way to go about this?

Thanks!
Patrick

**Comment by jofmi** [comment]
Hi Patrick,

glad to hear that the package is useful to you!

if you want the agents to be able to switch type, i would recommend to model them as the same class and make the algorithms dependent on the type attribute. 

Best,
Joël
